### PR TITLE
feat(query): 🔥 add `injectIsMutating` and `injectIsFetching` 

### PR DESCRIPTION
**Pull request by luii**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/query/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: query/src/lib/is-fetching.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    },
+  }
+);
+
+export function injectIsFetching(toSignal: boolean = false) {
+  return inject(UseIsFetching);
+}
+
+const isFetching = injectIsFetching();
+isFetching()
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts]: ?

File: query/src/lib/is-fetching.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  {
+    providedIn: 'root',
+    factory() {
+      const isFetching = new IsFetchingService();
+      return isFetching.use.bind(isFetching);
+    },
+  }
+);
+
+export function injectIsFetching(toSignal: boolean = false) {
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts]: Let's make the options an object

File: query/src/lib/is-fetching.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { QueryFilters, notifyManager } from '@tanstack/query-core';
+import { injectQueryClient } from './query-client';
+import { inject, Injectable, InjectionToken } from '@angular/core';
+import { distinctUntilChanged, Observable } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class IsFetchingService {
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts:7]:
```suggestion
export class IsFetching {
```

File: query/src/lib/is-fetching.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  {
+    providedIn: 'root',
+    factory() {
+      const isFetching = new IsFetchingService();
+      return isFetching.use.bind(isFetching);
+    },
+  }
+);
+
+export function injectIsFetching() {
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts:33]: We should also support signal. Two options I can think of:
```
export function injectIsFetching({ asSignal: true }) { }
export function injectIsFetching(): { toSignal(), result$ } { 
}
```

File: query/src/lib/is-fetching.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { distinctUntilChanged, Observable } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class IsFetchingService {
+  private queryClient = injectQueryClient();
+
+  use(filters?: QueryFilters) {
+    return new Observable<number>((observer) => {
+      observer.next(this.queryClient.isFetching(filters));
+      this.queryClient.getQueryCache().subscribe(
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts:13]: We need to unsubscribe. I see that we didn't do that on v1 :(

File: query/src/lib/is-mutating.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import {
+  type MutationFilters,
+  notifyManager
+} from '@tanstack/query-core';
+import { injectQueryClient } from './query-client';
+import { inject, Injectable, InjectionToken } from '@angular/core';
+import { distinctUntilChanged, Observable } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class IsMutatingService {
```
Comment (NetanelBasal) [inline query/src/lib/is-mutating.ts:10]: same comments here

File: query/src/lib/is-fetching.ts (hunk 1 above)
Comment (luii) [reply query/src/lib/is-fetching.ts]: i was just testing the typing there, but removed it in the last commit, just forgot to remove this

File: query/src/lib/is-fetching.ts (hunk 4 above)
Comment (luii) [reply query/src/lib/is-fetching.ts:45]: I've used the latter approach, if you think that it fits, than resolve the conversations

File: query/src/lib/is-fetching.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { QueryFilters, notifyManager } from '@tanstack/query-core';
+import { injectQueryClient } from './query-client';
+import { DestroyRef, inject, Injectable, InjectionToken } from '@angular/core';
+import { distinctUntilChanged, Observable } from 'rxjs';
+import { takeUntilDestroyed, toSignal } from '@angular/core/rxjs-interop';
+
+@Injectable({ providedIn: 'root' })
+export class IsFetching {
+  private queryClient = injectQueryClient();
+  private destroyRef = inject(DestroyRef);
```
Comment (NetanelBasal) [inline query/src/lib/is-fetching.ts:10]: destroyRef is redundant here because it's providedIn: 'root'. Let's remove it and let the consumer handle this.

**Comment by NetanelBasal** [comment]
Approved, can you please update the node version in our CI:

https://github.com/ngneat/query/actions/runs/6760534197/job/18374418728?pr=118

File: query/src/lib/is-fetching.ts (hunk 5 above)
Comment (luii) [reply query/src/lib/is-fetching.ts]: I think this could be also resolved then since im disposing the internal subscription on tearing down the observable

**Pull request merged**